Added support for outputting Javascript in serialized JSON

MarcoMiltenburg opened the pull request:
I had the need to output Javascript code in my JSON. You might wonder why... Well, for example, the JSON that must be passed in to a dataLayer push for Google Tag Manager supports callback functions:

``` Javascript
<script>
/**
 * Call this function when a user clicks on a product link. This function uses the event
 * callback datalayer variable to handle navigation after the ecommerce data has been sent
 * to Google Analytics.
 * @param {Object} productObj An object representing a product.
 */
function(productObj) {
  dataLayer.push({
    'event': 'productClick',
    'ecommerce': {
      'click': {
        'actionField': {'list': 'Search Results'},      // Optional list property.
        'products': [{
          'name': productObj.name,                      // Name or ID is required.
          'id': productObj.id,
          'price': productObj.price,
          'brand': productObj.brand,
          'category': productObj.cat,
          'variant': productObj.variant,
          'position': productObj.position
         }]
       }
     },
     'eventCallback': function() {
       document.location = productObj.url
     }
  });
}
</script>
```
In this case `'eventCallback'` must be set to "raw" Javascript code. I added a JSONscript object that allows this. You set it's `value` property to a string with the code you want to insert and it will serialize that string as raw output.

I've also added a test case to `test.asp` that also demonstrates it's usage.

MarcoMiltenburg committed:
Added support for JSONscript to output raw javascript to outputted JSON.

MarcoMiltenburg committed:
Added test to test if javascript it outputted correctly.

MarcoMiltenburg committed:
Accidentally forgot to remove 'dim' while cleaning up.

rcdmk commented on jsonObject.class.asp:
```diff
@@ -1114,6 +1117,62 @@ class JSONarray
+	public function Serialize()
+		dim js, out, instantiated, actualLCID
+		
+		actualLCID = Response.LCID
```
We don't need to manage LCID in this context. This is only needed when dealing with numbers and dates.

rcdmk commented on jsonObject.class.asp:
```diff
@@ -1114,6 +1117,62 @@ class JSONarray
+	end property
+	
+	public property let value(newValue)
+		if (len(newValue) = 0) then newValue = s_nullString
```
A type check would be better to ensure that only strings and `Null` are valid before checking the length.

rcdmk commented on jsonObject.class.asp:
```diff
@@ -1114,6 +1117,62 @@ class JSONarray
 
 
+class JSONscript
+	dim i_parent, i_depth, i_version
```
Depth and parent reference are not needed and can be removed.

MarcoMiltenburg replied:
I wasn't quite sure if they were used from the parent object, so that's why I left it in. But it's removed now.

rcdmk commented on jsonObject.class.asp:
```diff
@@ -1114,6 +1117,62 @@ class JSONarray
+	end property
+	
+	' The parent object
+	public property get parent
```
The depth and parent properties should also be removed.

rcdmk reviewed (changes requested):
Good! I have some suggestions to make it better.

MarcoMiltenburg committed:
Removed changing LCID as it's not needed.

MarcoMiltenburg committed:
Cleaned up and simplified code.

MarcoMiltenburg committed:
Added check on the type of the passed in new value.

MarcoMiltenburg committed:
Removed the parent property and i_depth variable as it's not needed.

MarcoMiltenburg committed:
Changed the version number to 1.0.0.
All classes are versioned independently, as per comment: https://github.com/rcdmk/aspJSON/pull/62#issuecomment-373964189

rcdmk reviewed (approved):
Great!
